log_outgoing_requests/log_requests.py

```diff
@@ -15,6 +17,18 @@ def hook_requests_logging(response, *args, **kwargs):
     logger.debug("Outgoing request", extra=extra)
 
 
+def hook_requests_exception(e, *args, **kwargs):
```

Comment:
this isn't a great name as it seems to imply that it makes use of requests hooks feature too, which is not the case. it also has a different signature